src/accelerate/accelerator.py

```diff
@@ -421,10 +427,30 @@ def __init__(
                 )
 
         self.device_placement = device_placement
-        self.split_batches = split_batches
-        self.dispatch_batches = dispatch_batches
-        self.even_batches = even_batches
-        self.use_seedable_sampler = use_seedable_sampler
+        if dataloader_config is None:
+            self.dataloader_config = DataLoaderConfiguration()
+        # Deal with deprecated args
+        deprecated_dl_args = {}
+        if dispatch_batches is not None:
+            deprecated_dl_args["dispatch_batches"] = dispatch_batches
```

Comment:
There is a bit of a danger that if a user passes `Accelerator(split_batches=True)` (i.e. passes the default), they will not see the deprecation message and their code will eventually fail. Two solutions come to mind:

- Remove the deprecated arguments from the `__init__` signature. Catch them with `kwargs` and check if they were passed explicitly.
- Use a sentinel value instead of the default and check if that value is being used:

```python
_split_batches_default = object()
...
def __init__(
    self,
    ...,
    split_batches=_split_batches_default,
):
    if split_batches is _split_batches_default:
        split_batches = True
    else:
        deprecated_dl_args["dispatch_batches"] = dispatch_batches
```

Both not very nice unfortunately.

Reply:
The first one is better I think here. Not great, yes, but an easier transition especially when dealing with defaults.

Reply:
Actually, the Accelerator doesn't have the ability to do `kwargs`. So we will have to do the first.